src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java

```diff
@@ -237,7 +237,7 @@ protected boolean mustCheckName(DetailAST ast) {
         if (ignoreOverridden && isOverriddenMethod(ast)
                 || parent.getType() == TokenTypes.LITERAL_CATCH
                 || parent.getParent().getType() == TokenTypes.LAMBDA
-                || CheckUtil.isReceiverParameter(ast)
+                || ast.findFirstToken(TokenTypes.IDENT) == null
```

Comment:
Everyone knows that parameter has ident.
But fact that `this ` is legal name with special meaning is unlikely on the top of people make mind.
Let's convert all to `ast.findFirstToken(TokenTypes.This_literal) != null` and `ast.findFirstToken(TokenTypes.This_literal) == null` to educate reader that `this` possible to happen.

Reply:
@romani This method is used by many check. and after make change `ast.findFirstToken(TokenTypes.This_literal) != null`
and evaluating the code this can be valid only when the code is 
```
    public void foo4(InputFinalParametersReceiver this) {}
```
The ast would be 
```
    |--METHOD_DEF -> METHOD_DEF [12:4]
    |   |--MODIFIERS -> MODIFIERS [12:4]
    |   |   `--LITERAL_PUBLIC -> public [12:4]
    |   |--TYPE -> TYPE [12:11]
    |   |   `--LITERAL_VOID -> void [12:11]
    |   |--IDENT -> foo4 [12:16]
    |   |--LPAREN -> ( [12:20]
    |   |--PARAMETERS -> PARAMETERS [12:21]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [12:21]
    |   |       |--MODIFIERS -> MODIFIERS [12:21]
    |   |       |--TYPE -> TYPE [12:21]
    |   |       |   `--IDENT -> InputFinalParametersReceiver [12:21]
    |   |       `--LITERAL_THIS -> this [12:50]
    |   |--RPAREN -> ) [12:54]
    |   `--SLIST -> { [12:56]
    |       `--RCURLY -> } [12:57]

```
In this type of code  `ast.findFirstToken(TokenTypes.This_literal) != null` logic is fine but 

```
        public Inner(InputFinalParametersReceiver InputFinalParametersReceiver.this) {}
```
ast 
```
    |       |   |--LPAREN -> ( [15:20]
    |       |   |--PARAMETERS -> PARAMETERS [15:21]
    |       |   |   `--PARAMETER_DEF -> PARAMETER_DEF [15:21]
    |       |   |       |--MODIFIERS -> MODIFIERS [15:21]
    |       |   |       |--TYPE -> TYPE [15:21]
    |       |   |       |   `--IDENT -> InputFinalParametersReceiver [15:21]
    |       |   |       `--DOT -> . [15:78]
    |       |   |           |--IDENT -> InputFinalParametersReceiver [15:50]
    |       |   |           `--LITERAL_THIS -> this [15:79]
    |       |   |--RPAREN -> ) [15:83]
    |       |   `--SLIST -> { [15:85]
    |       |       `--RCURLY -> } [15:86]

```
in this case we need the condition so only `ast.findFirstToken(TokenTypes.This_literal) != null` is not enough

thats why i have updated exisiting method

Reply:
code becomes hard to read, lets return to `parameterDefAst.findFirstToken(TokenTypes.IDENT) == null`
and lets keep util method for it - isReceiverParameter.
BUT please extend documentation for this method to explain what is Receiver (link to jdk spec, see it in my comments) and fact that receiver can be on few levels deeper.
and please
extend https://github.com/checkstyle/checkstyle/blob/8134680099d82c13c8231ca1b96048a01773a356/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java#L715 with `SomeType.this` to show what is Receiver and how it looks.
please put link to javadoc of PARAMETER_DEF in this method javadoc.


Sorry for misguiding in wrong direction.

Reply:
Done